torchci/components/benchmark/llms/LLMsBenchmarkPage.tsx

```diff
@@ -0,0 +1,349 @@
+import { Stack, Typography } from "@mui/material";
+import {
+  DEFAULT_REPO_NAME,
+  LAST_N_DAYS,
+  MAIN_BRANCH,
+} from "components/benchmark/common";
+import CopyLink from "components/CopyLink";
+import { Granularity } from "components/metrics/panels/TimeSeriesPanel";
+import dayjs from "dayjs";
+import _, { cloneDeep } from "lodash";
+import { NextRouter, useRouter } from "next/router";
+import { ParsedUrlQuery } from "querystring";
+import { useEffect, useReducer, useState } from "react";
+import { propsReducer } from "./context/BenchmarkProps";
+
+import LoadingPage from "components/LoadingPage";
+import {
+  DEFAULT_ARCH_NAME,
+  DEFAULT_BACKEND_NAME,
+  DEFAULT_DEVICE_NAME,
+  DEFAULT_DTYPE_NAME,
+  DEFAULT_MODE_NAME,
+  DEFAULT_MODEL_NAME,
+  REPO_TO_BENCHMARKS,
+} from "lib/benchmark/llms/common";
+import { LLMsBenchmarkProps } from "lib/benchmark/llms/types/dashboardProps";
+import { getBenchmarkDropdownFeatures } from "lib/benchmark/llms/utils/dashboardPickerUtils";
+import {
+  getLLMsBenchmarkPropsQueryParameter,
+  useBenchmarkPropsData,
+} from "lib/benchmark/llms/utils/llmUtils";
+import { LLMsDashboardPicker } from "./components/dashboardPicker/LLMsDashboardPicker";
+import { LLMsTimeRangePicker } from "./components/dashboardPicker/LLMsTimeRangePicker";
+import LLMsReport from "./components/LLMsReport";
+
+export default function LLMsBenchmarkPage() {
+  const router = useRouter();
+
+  // Set the default start and stop time to be the last N days when the page is loaded
+  const defaultStartTime = dayjs().subtract(LAST_N_DAYS, "day");
+  const defaultStopTime = dayjs();
+
+  const initialPropsState: LLMsBenchmarkProps = {
```

Comment:
Also curious as to the preference of the props as a object vs explicit, is it just to make it easier to pass around and selectively get values or is there another reason I'm not thinking of?

Reply:
nvm i see pr body